pkg/scanners/scannerv4/convert.go

```diff
@@ -186,55 +188,114 @@ func vulnerabilities(vulnerabilities map[string]*v4.VulnerabilityReport_Vulnerab
+				score.CvssScore = &storage.CVSSScore_Cvssv3{Cvssv3: cvssV3}
+				scores = append(scores, score)
+				// CVSS metrics has maximum two entries, one from NVD, one from Rox updater if available
+				if len(cvssMetrics) == 1 || (len(cvssMetrics) > 1 && cvss.Source != v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_NVD) {
+					vuln.CvssV3 = cvssV3.CloneVT()
+					// overwrite if v3 available
+					vuln.ScoreVersion = storage.EmbeddedVulnerability_V3
+					vuln.Cvss = baseScore
+					vuln.Link = cvss.Url
+				}
+			} else {
+				errList.AddError(v3Err)
 			}
-			c.Severity = cvssv2.Severity(c.Score)
-			vuln.CvssV2 = c
-			// This sets the top level score for use in policies.
-			// It will be overwritten if v3 exists.
-			vuln.ScoreVersion = storage.EmbeddedVulnerability_V2
-			vuln.Cvss = c.Score
-		} else {
-			errList.AddError(fmt.Errorf("v2: %w", err))
 		}
 	}
-	if v3 := vulnCVSS.GetV3(); v3 != nil {
-		if c, err := cvssv3.ParseCVSSV3(v3.GetVector()); err == nil {
-			err = cvssv3.CalculateScores(c)
-			if err != nil {
-				errList.AddError(fmt.Errorf("calculating CVSS v3 scores: %w", err))
-			}
-			// Use the report's score if it exists.
-			if baseScore := v3.GetBaseScore(); baseScore != 0.0 && baseScore != c.Score {
-				log.Debugf("Calculated CVSSv3 score does not match given base score (%f != %f) for %s. Using given score...", c.Score, baseScore, vuln.GetCve())
-				c.Score = baseScore
-			}
-			c.Severity = cvssv3.Severity(c.Score)
-			vuln.CvssV3 = c
-			// Overwrite v2, if set.
-			vuln.ScoreVersion = storage.EmbeddedVulnerability_V3
-			vuln.Cvss = c.Score
-		} else {
-			errList.AddError(fmt.Errorf("v3: %w", err))
+
+	if len(scores) > 0 {
+		vuln.CvssMetrics = scores
+		if errList.Empty() {
+			return nil
 		}
 	}
+
 	return errList.ToError()
 }
 
+func CVSSSource(source v4.VulnerabilityReport_Vulnerability_CVSS_Source) storage.Source {
+	switch source {
+	case v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_NVD:
+		return storage.Source_SOURCE_NVD
+	case v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_OSV:
+		return storage.Source_SOURCE_OSV
+	case v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_RED_HAT:
+		return storage.Source_SOURCE_RED_HAT
+	default:
+		return storage.Source_SOURCE_UNKNOWN
+	}
+}
+
+func toCVSSV2Scores(vulnCVSS *v4.VulnerabilityReport_Vulnerability_CVSS, cve string) (float32, *storage.CVSSV2, error) {
+	v2 := vulnCVSS.GetV2()
+	c, err := cvssv2.ParseCVSSV2(v2.GetVector())
+	if err == nil {
+		err = cvssv2.CalculateScores(c)
+		if err != nil {
+			return 0, nil, fmt.Errorf("calculating CVSS v2 scores: %w", err)
+		}
+		// Use the report's score if it exists.
+		if baseScore := v2.GetBaseScore(); baseScore != 0.0 && baseScore != c.Score {
+			log.Debugf("Calculated CVSSv2 score does not match given base score (%f != %f) for %s. Using given score...", c.Score, baseScore, cve)
+			c.Score = baseScore
+		}
+		c.Severity = cvssv2.Severity(c.Score)
+		return c.Score, c, nil
+	}
+	return 0, nil, err
```

Comment:
nit: `fmt.Errorf("parsing CVSS v2 vector: %w", err)` to keep it similar to `toCVSSV3Scores` behavior